docs/security/remediation/validate_ledger.py

```
@@ -0,0 +1,1029 @@
+
+
+def run_check(base_ref: str | None, check_render: bool) -> tuple[dict[str, Any], str]:
+    source_index = load_canonical(SOURCE_INDEX_PATH)
+    source_rows = validate_source_index(source_index)
+    ledger = load_canonical(FINDINGS_PATH)
+    head = current_head()
+    validate_findings(ledger, source_rows, head)
+    if base_ref:
+        validate_append_only(base_ref, ledger)
+    if check_render:
+        result = subprocess.run([sys.executable, str(RENDERER), "--check"], cwd=ROOT, check=False)
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)